### PR TITLE
Fixes #6408: Test and tools broken with new metadata format

### PR DESCRIPTION
**Pull request by VinceMacBuche**
http://www.rudder-project.org/redmine/issues/6408

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
PR updated

File: tests/unit/test_ncf.py [hunk 1]
```diff
@@ -174,16 +174,25 @@ def test_get_all_techniques_metadata(self):
     """get_all_techniques_metadata should return a list of all techniques with all defined metadata tags and methods_called"""
     metadata = ncf.get_all_techniques_metadata()
 
-    number = len(ncf.get_all_techniques_filenames())
-    self.assertEquals(number, len(metadata))
+    data = metadata["data"]
+    errors = metadata["errors"]
+
+    # Remove files in error
+    number = len(ncf.get_all_techniques_filenames()) - len(errors)
```
Comment (peckpeck) [inline tests/unit/test_ncf.py]: Are we sure that we should have only on error per file ?

File: tests/unit/test_ncf.py [hunk 2]
```diff
@@ -174,16 +174,25 @@ def test_get_all_techniques_metadata(self):
...
     alternative_path = os.path.dirname(os.path.realpath(__file__)) + "/test_methods"
     metadata = ncf.get_all_techniques_metadata(alt_path=alternative_path)
 
-    number = len(ncf.get_all_techniques_filenames(alternative_path))
-    self.assertEquals(number, len(metadata))
+    data = metadata["data"]
+    errors = metadata["errors"]
+
+    # Remove files in error
+    number = len(ncf.get_all_techniques_filenames(alternative_path)) - len(errors)
```
Comment (peckpeck) [inline tests/unit/test_ncf.py]: Idem

**Comment by VinceMacBuche** [comment]
@peckpeck  PR updated!

**Comment by peckpeck** [comment]
OK Good

**Pull request merged**